### PR TITLE
Basic support for hub-side tags

### PR DESCRIPTION
**Pull request by marmistrz**
API changes:

`/peer/{nodeId}/addTags`
accepts an `application/json` payload of form
```
["tag1", "tag2", "tag3"]
```
The following tags will be added to the set of the current tags. Tags which already have been added are ignored.

`/peer/{nodeId}/deleteTags`
accepts a `application/json` payload of form
```
["tag1", "tag2", "tag3"]
```
The following tags will be removed from the set of the current tags, if they exist. Non-existent tags are ignored.

Tasks:
* [x] setting tags
* [x] tag persistence
* [x] cleanup

### CODE REVIEW

**Review by filipgolem — APPROVED** [review]
Looks good to me. Please merge with release/0.2 and check if it works with --user and -c options which change the working dir.

File: gu-net/src/rpc/peer.rs [hunk 1]
```diff
@@ -62,16 +68,54 @@ impl Message for UpdatePeer {
...
 
 impl Actor for PeerManager {
     type Context = Context<Self>;
+
+    fn started(&mut self, _ctx: &mut <Self as Actor>::Context) {
+        self.path = ConfigModule::new().work_dir().join("tags");
+        // FIXME this one is awful
+        let tags_serialized = fs::read_to_string(&self.path).unwrap_or_else(|e| match e.kind() {
+            io::ErrorKind::NotFound => "{}".into(),
+            _ => panic!(
```
Comment (filipgolem) [inline gu-net/src/rpc/peer.rs:91]: What's better? 1) panic!(...) or 2) error!(..) and continue.

File: gu-net/src/rpc/peer.rs [hunk 2]
```diff
@@ -130,3 +177,91 @@ impl Handler<GetPeer> for PeerManager {
...
+    pub node: NodeId,
+    pub tags: Tags,
+}
+
+impl Message for AddTags {
+    type Result = Option<()>;
+}
+
+impl Handler<AddTags> for PeerManager {
+    type Result = Option<()>;
```
Comment (filipgolem) [inline gu-net/src/rpc/peer.rs]: Maybe Result<()>?

File: gu-net/src/rpc/peer.rs [hunk 3]
```diff
@@ -130,3 +177,91 @@ impl Handler<GetPeer> for PeerManager {
...
+    fn handle(&mut self, msg: AddTags, ctx: &mut Self::Context) -> Self::Result {
+        let mut peer = self.peers.get_mut(&msg.node)?;
+        let mut tags = &mut peer.tags;
+        tags.extend(msg.tags);
+        Some(())
```
Comment (filipgolem) [inline gu-net/src/rpc/peer.rs]: We can't change the tags of a disconnected provider.

File: gu-net/src/rpc/peer.rs [hunk 4]
```diff
@@ -130,3 +177,91 @@ impl Handler<GetPeer> for PeerManager {
...
+    pub node: NodeId,
+    pub tags: Tags,
+}
+
+impl Message for DeleteTags {
+    type Result = Option<()>;
+}
+
+impl Handler<DeleteTags> for PeerManager {
+    type Result = Option<()>;
```
Comment (filipgolem) [inline gu-net/src/rpc/peer.rs]: See previous comment

File: gu-net/src/rpc/peer.rs (hunk 2 above)
Comment (marmistrz) [reply gu-net/src/rpc/peer.rs]: I changed it to `Result<(), ()>` but this is not really expressive, but I'm not sure if I should add a new type just to convey "Peer not found". On the other hand, `Result<(), String>` doesn't seem clean.

File: gu-net/src/rpc/peer.rs (hunk 3 above)
Comment (marmistrz) [reply gu-net/src/rpc/peer.rs]: Fixed.

File: gu-net/src/rpc/peer.rs (hunk 4 above)
Comment (marmistrz) [reply gu-net/src/rpc/peer.rs]: Fixed.

File: gu-net/src/rpc/peer.rs (hunk 1 above)
Comment (marmistrz) [reply gu-net/src/rpc/peer.rs]: `panic!` - if there are some other problems with permissions/filesystem, they should be fixed by the administrator, or users will start getting problems.

**Pull request closed without merging**